File: frontend/src/api/copilot.ts
import apiClient from './client';

export interface CopilotResult {
  answer: string;
  chart: { title: string; chart_type: 'bar' | 'line' | 'pie'; data: Array<{ bucket: string; incidents: number }> } | null;
  incidents: Array<{ id: string; event_type: string; timestamp: string; camera_name: string; zone_name: string; reviewed: boolean; has_clip: boolean; clip_url: string | null }>;
}

export interface CopilotHistoryMessage { role: 'user' | 'model'; text: string; }

export const copilotApi = {
  ask: async (message: string, history: CopilotHistoryMessage[]): Promise<CopilotResult> => (await apiClient.post('/copilot/chat', { message, history })).data,
};

File: backend/app/schemas/copilot.py
from typing import Literal

from pydantic import BaseModel, Field


class CopilotHistoryMessage(BaseModel):
    role: Literal["user", "model"]
    text: str = Field(..., min_length=1, max_length=4000)


class CopilotRequest(BaseModel):
    message: str = Field(..., min_length=1, max_length=2000)
    history: list[CopilotHistoryMessage] = Field(default_factory=list, max_length=12)
